cluster/sbin/snappy-nodes.sh

```diff
@@ -338,12 +346,44 @@ function getNumLeadsOnHost() {
   echo $numLeadsOnHost
 }
 
+function scpConf() {
+  currentNodeIpAddr=$(ip addr | grep 'state UP' -A2 | head -n3 | tail -n1 | awk '{print $2}' | cut -f1  -d'/')
+  currentNodeHostName=$(uname -n)
+
+  if [[ "$host" != "$currentNodeIpAddr" && "$host" != "localhost" && "$host" != $currentNodeHostName ]] ; then
+  #loop to get the all the files avaliable in Conf directory
+    for entry in "${SPARK_CONF_DIR}"/*; do
+      if [ -f "$entry" ];then
+        #${file%.*} to get the filename without the extension and ${file##*.} to get the extension alone
+	fileName=$(basename $entry)
+ 	template=".template"
+	#check the extension, interested in files those doesn't have template extension
+	if [[ ! "$fileName" = @(*.template) ]]; then	       	
+	  if ! ssh $host "test -e $entry"; then #"File does not exist."	  
+            scp ${SPARK_CONF_DIR}/$fileName  $host:${SPARK_CONF_DIR}
+	  else
+	      backupDir="backup"
+	      if [[ ! -z $(ssh $host "cat $entry" | diff - "$entry") ]] ; then
+		backupFileName=${fileName}_`date +%Y_%m_%d_%H_%M_%S`
```

Comment:
Can initialize it once and use it for each execute() call.

Reply:
Added timestamp in snappy-start-all.sh